### ISSUE TITLE
Allow AWS Script step to use custom versions of powershell modules and AWS CLI

### ISSUE THREAD

**Issue by shaunmarx** [OctopusDeploy/Issues#4964]
# Prerequisites

- [x] I have searched [open](https://github.com/OctopusDeploy/Issues/issues) and [closed](https://github.com/OctopusDeploy/Issues/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed) issues to make sure it isn't already requested
- [x] I have written a descriptive issue title
- [x] I have linked the original source of this feature request
- [x] I have tagged the issue appropriately (area/*, kind/feature)

# The feature

Allow AWS Script step to specify a custom versions of the AW Powershell modules and the AWS CLI

## What this feature isn't

The feature would not cover our dependency on the AWS SDK. 

## Elevator pitch for this feature

Providing no means to override the modules/CLI forces customers to use these specific versions. This effectively means there are no workarounds if a customer runs into an issue. It also means customers have to wait for us to update our dependencies to potentially gain access to new features.

## How much noise should we make about this?

None. This is to bring the AWS Script step in-line with the Azure script step.

# Links

https://help.octopus.com/t/deployments-using-powershell-script-fail-when-disabling-tls-1-0-on-destination-environment/21475

**Comment by matt-richardson** [comment]
In theory, superseded by our planned support for "tools" packages, but adding as a quick win for now to unblock customers until that point.

**Comment by octoreleasebot** [comment]
Release Note: Added special variable `OctopusUseBundledAwsCLI` to allow opting out of using the bundled AWS CLI
